src/cml.js

```diff
@@ -387,7 +387,7 @@ class CML {
           log.job = parseId('job');
           log.pipeline = parseId('pipeline');
 
-          if (name && this.driver === GITHUB) {
+          if (name && cloudSpot && this.driver === GITHUB) {
```

Comment:
```suggestion
          // GitHub API doesn’t seem to provide a straightforward way to get the
          // job identifier from the runner logs, so we need to query several API
          // endpoints to retrieve it. Due to several broken parts of our logic and
          // some unexpected API responses, performing these queries may trigger API
          // rate limits, causing the whole `cml` process to crash. Given that
          // retrieving the job identifier is only useful for spot instance recovery
          // (i.e. automated retryWorkflow), we can save ourselves all the hassle if
          // —-cloud-spot is not set or —-driver is not GitHub.
          if (cloudSpot && this.driver === GITHUB) {
```

Something like that? Also removed the `ìf (name)` condition because it’s a tautology,

Reply:
A bit verbose, but still much clearer - thanks!